Update CDK doc links, install instructions

rkratky opened the pull request:
Please merge updates of CDK 2.1 doc links (the Docs and APIs section) and installation instructions (the Get Started section).

JBossJenkins commented:
Can one of the admins verify this patch?

RubyGemTSL commented:
this is ok to test

RubyGemTSL commented:
tests will fail due to failing tests being pushed over the weekend. I will have a PR merged soon to fix this.

LightGuard commented:
retest this please

rkratky commented:
I accidentally added commits from PR #1228 to this PR. Feel free to remove the last three commits from this PR.

rkratky commented:
Is there any way I can help to speed this up? It's rather unfortunate that old docs are still referenced from the Dev Portal.

paulrobinson commented:
Apologies, it looks like this has fallen through the cracks. There's a lot
of stuff going on right now, with it being Summit week.

I'll take a look once I next get access to my laptop. Please remind me if
this gets forgotten again.

On 29 Jun 2016, at 10:17, rkratky <notifications@github.com> wrote:

Is there any way I can help to speed this up? It's rather unfortunate that
old docs are still referenced from the Dev Portal.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/redhat-developer/developers.redhat.com/pull/1225#issuecomment-229425266>,
or mute the thread
<https://github.com/notifications/unsubscribe/AADJ4fi57P8oYy81oY6Rhxjj1prYXNumks5qQqifgaJpZM4I_S_a>
.

LightGuard commented:
@paulrobinson Pretty sure this was forgotten again... I don't know if it's good to merge or not.

rkratky commented:
Bumping this again. Are there any problems with the PR?

rctrh commented on products/cdk/get-started-cdk2-ruby.adoc:
```diff
@@ -64,7 +64,7 @@ include::products/cdk/_includes/get-started-cdk2-step2b.adoc[]
 In this step, you will create a tiny Hello World container that uses {tthw-langshort} as a web server. Once created, the container can be run on other systems that have `docker` installed.  You will need to create several files in an empty directory using your favorite editor, including a `Dockerfile` that describes how to build the container image.
 
-Note: you can edit files on your host system that can be synchronized to your vagrant box with `vagrant rsync`. For more information see _Vagrant synchronizad folders_ in the _CDK Installation Guide_. 
+Note: When you use SSHFS, your home directory from the host system is automatically synchronized to your Vagrant box. For more information, see https://access.redhat.com/documentation/en/red-hat-container-development-kit/2.1/getting-started-guide/#synced-folders[Vagrant Synchronizad Folders] in the _CDK Getting Started Guide_. 
```
FYI, There are 4 files here with the same wording covering Node, PHP, Python, and Ruby to keep in sync.  This commit hits only two. 

With this edit SSHFS seems a little out of context.  Suggest something like:

You can edit files on your host system and have them automatically synchronized to your Vagrant box using SSHFS. See https://access.redhat.com/documentation/en/red-hat-container-development-kit/2.1/getting-started-guide/#synced-folders[Vagrant Synchronizad Folders] in the _CDK Getting Started Guide_.

rctrh commented:
Did the build fail or has it been already been cleaned up?   Can an admin please trigger a new build.

Looking at the commits, without checking the preview:

CDK commits
* The first two commits f0b2219 and fc2ff76 look good to me.
* I added a comment on 94f2113, 1) there are 4 language TTHWs that should be in sync. 2) suggested a wording change.

What's being done with the DTS commits are they going to get merged with this PR or reverted?   I spot checked the diffs, they look fine.

paulrobinson commented:
@rkratky I think the problem is that we need someone to do a technical review of the content. Someone, ideally not the PR creator, needs to read over the changes made (via the preview build) and check the content ins correct and that the links do what is expected. The engineering team focuses on checking that the code is sound and that it doesn't break anything. We are not domain experts so we can't check the content. 

We merged a bunch of things pre-summit that didn't get a thorough content review. This resulted in *many* broken things on the site and *many* JIRAs and emails to me pointing out all the problems. We can't let this happen again. 

@rkratky Can you suggest a person to do the content review? You might need multiple people, as I notice the PR covers many products.

paulrobinson commented:
@rkratky FYI: I'm on PTO this week, so please try and resolve this with the rest of my team.

rkratky commented:
@rctrh, I modified the SSHFS wording as per your suggestion and added the same note to the other two TTHW files (which I overlooked before -- thanks for pointing that out).

@paulrobinson, I'll try to find people to review the changes. The PR covered two products because I accidentally pushed the changes for the second product to the same branch as the first ones. I've now reverted those commits to keep it organized. There's a separate PR (#1228) for the changes to the second product.

pmuir commented on products/cdk/get-started-cdk2-nodejs.adoc:
```diff
@@ -63,7 +63,7 @@ include::products/cdk/_includes/get-started-cdk2-step2b.adoc[]
 In this step, you will create a tiny Hello World container that uses {tthw-langshort} as a web server. Once created, the container can be run on other systems that have `docker` installed.  You will need to create several files in an empty directory using your favorite editor, including a `Dockerfile` that describes how to build the container image.
 
-Note: you can edit files on your host system that can be synchronized to your vagrant box with `vagrant rsync`. For more information see _Vagrant synchronizad folders_ in the _Container Development Kit Installation Guide_. 
+Note: You can edit files on your host system and have them automatically synchronized to your Vagrant box using SSHFS. See https://access.redhat.com/documentation/en/red-hat-container-development-kit/2.1/getting-started-guide/#synced-folders[Vagrant Synchronizad Folders] in the _CDK Getting Started Guide_. 
```
Just wonder if we want to make this technology-agnostic. IOW the team are investigating other methods of syncing folders as well that may work better - so remove "using SSHFS".

rkratky replied:
Good idea -- the Get Started on dev.rh.com doesn't need to have the implementation details. I'll push a correction.

pmuir commented on products/cdk/get-started.adoc:
```diff
@@ -59,25 +59,25 @@ First, download and install a virtualization platform:
 
 [.callout-light]
-*NOTE:* We recommend you use Vagrant version 1.7.4, or a version *later than 1.8.1* on Microsoft Windows and Apple Mac OS X. Vagrant releases 1.8.0 and 1.8.1 should be avoided due to some bugs that impact proper operation of the kit.
+*NOTE:* Use Vagrant version 1.7.4 with VirtualBox on Microsoft Windows. Use version 1.8.1 and higher with Hyper-V. On Red Hat Enterprise Linux, use Vagrant 1.7.4.
```
BTW @LalatenduMohanty hferentschik@ I had problems with latest Windows 10 updates and vagraqnt 1.7.4 (and I had to update virtual box). I'm running Windows Insider (so I get previews) but I suspect we need to be on top of this for when they update windows...

rkratky replied:
This is based on QE findings for CDK 2.1. I can update this when new testing gets done.

pmuir commented on products/cdk/get-started.adoc:
```diff
@@ -90,7 +90,7 @@ To provide `ssh` and `rsync` on Microsoft Windows, install http://cygwin.com/ins
 
 
-### Download the Container Runtime Environment
+### Download the Container Development Environment
```
I thought we went for CRE in the end?

rkratky replied:
Not AFAIK. See https://docs.google.com/document/d/17hbzJY2eV4dwjKO2p_4mOh3Xqno9LgUobOnTHmp4iMo/edit

rkratky commented:
Pushed changes based on Pete's suggestions and a couple of other fixes.

RickJWagner commented:
I've validated Robert's recent changes, they correctly fulfill an outstanding JIRA.  (Please allow this PR).
